core/src/apps/zcash/unified_addresses.py

```diff
@@ -0,0 +1,118 @@
+"""
+Implementation of encoding and decoding of Zcash
+unified addresses according to the ZIP-316.
+see: https://zips.z.cash/zip-0316
+"""
+
+from typing import Dict
+
+from trezor.crypto.bech32 import Encoding, bech32_decode, bech32_encode, convertbits
+from trezor.enums import ZcashReceiverTypecode as Typecode
+from trezor.utils import BufferReader, empty_bytearray
+from trezor.wire import DataError
+
+from apps.common.coininfo import CoinInfo
+from apps.common.readers import read_compact_size
+from apps.common.writers import write_bytes_fixed, write_compact_size
+
+from .f4jumble import f4jumble, f4unjumble
+
+
+def receiver_length(typecode: int):
+    """Byte length of a receiver."""
+    if typecode in [Typecode.P2PKH, Typecode.P2SH]:
```

Comment:
nit: these should be tuples `(Typecode.P2PKH, Typecode.P2SH)` not lists

Reply:
[c536e71](https://github.com/trezor/trezor-firmware/pull/2398/commits/c536e71da57fdbcde00ff0c87405f3f51b4a25bf)